### PR TITLE
PYTHON-4076 [pymongocrypt] Use PyPI Trusted Publishing

### PR DESCRIPTION
**Pull request by blink1073**
Wheel build example: [blink1073/libmongocrypt/actions/runs/7187206572?pr=3](https://github.com/blink1073/libmongocrypt/actions/runs/7187206572?pr=3)

### CODE REVIEW

File: .github/workflows/release-python.yml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  run:
+    working-directory: ./bindings/python
+    shell: bash -eux {0}
+
+jobs:
+  build_dist:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-20.04, windows-latest, macos-latest]
```
Comment (ShaneHarvey) [inline .github/workflows/release-python.yml:26]: Why use windows-latest and macos-latest vs ubuntu-20.04? Should we pin the windows/mac hosts too?

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]
Need to address the failing EVG release tasks

**Comment by blink1073** [comment]
Updated run with Ubuntu-latest: https://github.com/blink1073/libmongocrypt/actions/runs/7189759310?pr=3

File: .github/workflows/release-python.yml (hunk 1 above)
Comment (blink1073) [reply .github/workflows/release-python.yml]: Works with ubuntu-latest

**Comment by blink1073** [comment]
> Need to address the failing EVG release tasks

Done

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]
The EVG release process generates these files:
```
$ ls -l /Users/shane/Downloads/libmongocrypt_python_release_release_python_combine_patch_b3d65b3aefb024c2cdf4fd65397c2ec28ac89ede_65791b3161837d6b5dffbd8c_23_12_13_02_47_14-0-release-files-all
total 56360
-rw-r--r--@ 1 shane  staff  3469786 Dec 12 18:47 pymongocrypt-1.7.1.dev0-py3-none-macosx_10_14_universal2.whl
-rw-r--r--@ 1 shane  staff  3469788 Dec 12 18:47 pymongocrypt-1.7.1.dev0-py3-none-macosx_10_14_x86_64.whl
-rw-r--r--@ 1 shane  staff  6902650 Dec 12 18:48 pymongocrypt-1.7.1.dev0-py3-none-macosx_11_0_universal2.whl
-rw-r--r--@ 1 shane  staff  6902646 Dec 12 18:48 pymongocrypt-1.7.1.dev0-py3-none-macosx_11_0_x86_64.whl
-rw-r--r--@ 1 shane  staff  5369900 Dec 12 18:57 pymongocrypt-1.7.1.dev0-py3-none-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
-rw-rw-r--@ 1 shane  staff  2671752 Dec 12 18:48 pymongocrypt-1.7.1.dev0-py3-none-win_amd64.whl
-rw-r--r--@ 1 shane  staff    48884 Dec 12 18:48 pymongocrypt-1.7.1.dev0.tar.gz
```
Whereas the github one generates these:
```
$ ls -l /Users/shane/Downloads/all-dist-PYTHON-4076-2
total 29304
-rw-r--r--@ 1 shane  staff  6902646 Dec 13 02:05 pymongocrypt-1.7.1.dev0-py3-none-macosx_11_0_x86_64.whl
-rw-r--r--@ 1 shane  staff  5369899 Dec 13 02:05 pymongocrypt-1.7.1.dev0-py3-none-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
-rw-r--r--@ 1 shane  staff  2671960 Dec 13 02:05 pymongocrypt-1.7.1.dev0-py3-none-win_amd64.whl
-rw-r--r--@ 1 shane  staff    48827 Dec 13 02:05 pymongocrypt-1.7.1.dev0.tar.gz
```

So we're missing 3 macos wheels.

**Comment by ShaneHarvey** [comment]
Actually I was looking at the EVG release task from this PR. The current EVG job generates these set of files which we should maintain ([see here](https://spruce.mongodb.com/task/libmongocrypt_python_release_release_python_combine_48243fce4a9c784e5f0a30125c6729e88eac1376_23_12_13_14_04_11/files?execution=0)):
```
$ ls -l /Users/shane/Downloads/libmongocrypt_python_release_release_python_combine_48243fce4a9c784e5f0a30125c6729e88eac1376_23_12_13_14_04_11-0-release-files-all
total 45080
-rw-r--r--@ 1 shane  staff  3465910 Dec 13 09:30 pymongocrypt-1.7.1.dev0-py3-none-macosx_10_14_x86_64.whl
-rw-r--r--@ 1 shane  staff  6898773 Dec 13 09:30 pymongocrypt-1.7.1.dev0-py3-none-macosx_11_0_universal2.whl
-rw-r--r--@ 1 shane  staff  4994726 Dec 13 10:02 pymongocrypt-1.7.1.dev0-py3-none-manylinux2010_x86_64.whl
-rw-r--r--@ 1 shane  staff  4995092 Dec 13 10:04 pymongocrypt-1.7.1.dev0-py3-none-manylinux_2_12_x86_64.manylinux2010_x86_64.whl
-rw-rw-r--@ 1 shane  staff  2667869 Dec 13 09:36 pymongocrypt-1.7.1.dev0-py3-none-win_amd64.whl
-rw-r--r--@ 1 shane  staff    44639 Dec 13 09:30 pymongocrypt-1.7.1.dev0.tar.gz
```

**Comment by blink1073** [comment]
We can't support RHEL 6 anymore, so we'll need to use manylinux2014.

**Comment by blink1073** [comment]
Here's the [latest](https://github.com/blink1073/libmongocrypt/actions/runs/7203870273?pr=3):

```
pymongocrypt-1.7.1.dev0-py3-none-macosx_10_14_x86_64.whl
pymongocrypt-1.7.1.dev0.tar.gz
pymongocrypt-1.7.1.dev0-py3-none-win_amd64.whl
pymongocrypt-1.7.1.dev0-py3-none-manylinux_2_17_x86_64.manylinux2014_x86_64.whl
pymongocrypt-1.7.1.dev0-py3-none-macosx_11_0_universal2.whl
```

**Comment by ShaneHarvey** [comment]
> We can't support RHEL 6 anymore, so we'll need to use manylinux2014.

Can you elaborate? What's the problem with using manylinux2010 like we do currently?

**Comment by blink1073** [comment]
We were using the rhel6 build of libmongocrypt, but those EVG hosts are marked for removal.  https://github.com/mongodb/libmongocrypt/blob/5bc7d1bb55fa653dfce7ae7f219857cc5fd90443/bindings/python/release.sh#L70

**Comment by blink1073** [comment]
The RHEL 7 version of libmongocrypt has incompatible symbols with manylinux2010.

**Comment by ShaneHarvey** [comment]
Okay that sounds fine then. Could you update the changelog to mention the change from 2010 to 2014?

**Comment by blink1073** [comment]
> Okay that sounds fine then. Could you update the changelog to mention the change from 2010 to 2014?

Done.

**Review by ShaneHarvey — APPROVED** [review]
LGTM

**Pull request merged**